app/models/site.rb

```diff
@@ -0,0 +1,7 @@
+class Site < ActiveRecord::Base
+  validates :application_name, presence: true, allow_nil: true
```

Comment:
can you explain what this validator does? How is it simultaneously validating presence and allowing nil?